pkg/operator/controller/ingress/status.go

```diff
@@ -171,35 +181,109 @@ func computeDeploymentPodsScheduledCondition(deployment *appsv1.Deployment, pods
 }
 
 // computeIngressAvailableCondition computes the ingress controller's current Available status state
-// by inspecting the Available condition of deployment. The ingresscontroller is only available if
-// the deployment is also available.
-func computeIngressAvailableCondition(deployment *appsv1.Deployment) operatorv1.OperatorCondition {
-	for _, cond := range deployment.Status.Conditions {
-		if cond.Type != appsv1.DeploymentAvailable {
-			continue
+// by inspecting the following:
+// 1) the Available condition of deployment
+// 2) the DNSReady condition
+// 3) the LoadBalancerReady condition
+// The ingresscontroller is judged Available only if all 3 conditions are true
+func computeIngressAvailableCondition(conditions []operatorv1.OperatorCondition) []operatorv1.OperatorCondition {
+	conditionsMap := make(map[string]*operatorv1.OperatorCondition)
+	for i := range conditions {
+		conditionsMap[conditions[i].Type] = &conditions[i]
+	}
+	expected := []expectedCondition{
+		{
+			condition: string(appsv1.DeploymentAvailable),
```

Comment:
You're checking the IngressController's status conditions, not the Deployment's, so you need to use the IngressController condition that mirrors the Deployment's, right?  
```suggestion
			condition: IngressControllerDeploymentAvailableConditionType,
```